### DISCUSSION TITLE
Callback example

### DISCUSSION THREAD

**Discussion by devendranaga** [tokio-rs/tokio#5269] [General]
Hello,

How do i use tokio to register and perform callbacks when there is timer expiry or socket events? The callback function may be in a different `impl` of a different file. And there can be many callbacks as possible.

Thanks,
Dev

**Comment by Darksonn** [comment]
To run some code once a timer expires, you can do this.
```Rust
tokio::spawn(async {
    tokio::time::sleep(my_duration).await;
    // run your code here
});
```
As for sockets, you can similarly use an `.await` on the appropriate socket method to wait for a socket event.

**Reply by devendranaga** [reply]
Ok, but should i be calling this like below?

```rust
impl Event {
    tokio::spawn(async {
        tokio::time::sleep(my_duration).await;
        // handle event
   }
}
```
Does this work?

**Reply by Noah-Kennedy** [reply]
No, that is not valid Rust code. tokio::spawn is a function which spawns a task on the runtime. You need to call it within another function.
